### ISSUE TITLE
[a11y]: [Tooltip] - Tooltip's Label is confusing,  Causing Accessibility Issues

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16868]
### Package

@carbon/react

### Browser

Safari, Firefox

### Package version

v1.58.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

Screen reader directly announces the tooltip content instead of the label upon focus via keyboard navigation. 

Provide a short, descriptive label that clarifies the component's function. This label should be announced by the screen reader before the tooltip content when the tooltip receives focus.

**Recommended Fix:**
Provide a short, descriptive label that clarifies the component's function. This label should be announced by the screen reader before the tooltip content when the tooltip receives focus. 

**Example:**
Consider a tooltip used for rating. The label could be "Rating" and the tooltip content could be "Click to select a rating." By announcing the label first, the screen reader user understands the context of the subsequent tooltip information.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tooltip--playground

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by Regine-Lambrecht** [comment]
This is still not solved. The `aria-labelledby` refering to the `span role:tooltip` overwrites the `button label`. The accessible name does therefore not contain the visible label, which is against 2.5.3 Label in Name.

**Comment by chrisgrimley** [comment]
Hi @2nikhiltom and @Regine-Lambrecht. I just picked this up while I was looking at tickets with 'needs accessibility info'. Assuming we are talking about tooltip on buttons, I think this is a misuse of the component. The button text should be sufficient to describe behaviour without the need for additional text. I imagine that Tooltip was originally used on icon buttons to convey more information, but now they're attached to a button (as you rightly highlighted @Regine-Lambrecht), aria-label on the Tooltip replaces the button text. My suggestion would be to reserve tooltips for icon buttons, but I'd be interested to hear what important use cases there might be for having it on buttons.

fyi @tay1orjones

**Comment by Regine-Lambrecht** [comment]
Yes, I refer to a button with visible label and a tooltip, like in the [Tooltip component](https://react.carbondesignsystem.com/?path=/story/components-tooltip--alignment):

<img width="2134" height="1162" alt="Image" src="https://github.com/user-attachments/assets/ee10ec88-1fb0-4e70-a1c2-674c83544c2a" />

**Comment by chrisgrimley** [comment]
Hi @Regine-Lambrecht, I just wanted to circle back to see if anything else is required from the Accessibility team on this one. As mentioned above, I think this behaviour is a hang-over from it's correct use on icon buttons where the aria-labelledby would have been appropriate. But on buttons with text, that aria attribute just replaces the button's name entirely. The quick fix here would be to use aria-describedby instead which will announce the button's role and name first and then the tooltip description. 

`<button aria-describedby="tooltip-id">This button has a tooltip</button>`
